### DISCUSSION TITLE
Keep cursor in same line when click on line number to select entire line

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#4150] [Ideas]
When you have "Line Numbers" visible, you can click on that line number to select the content on that entire line.
However, when you do this currently, it ends up selecting that entire line, but then it places the cursor on the next as well as the line highlight color if your theme has a color set for this.
There is probably a reason for this but to me it just seems a little visually confusing.

Suggesting maybe can add a setting to allow the cursor to end up at the end of that line numbers content, rather than placing the cursor on the following line.

You can also drag your left mouse button on multiple line numbers down to select multiple lines, which does the same where it moves the cursor to the next line number not clicked on.
If you try the opposite way, where you drag up to lines above, then it leaves the cursor at the beginning of the last line number you drag to.

Again this behavior may make sense the way that it is, but can just be a little visually confusing with the cursor and the line highlight color advancing to the next line.
Thank You

**Comment by Skif-off** [comment]
Does a triple mouse click work with the same result?

> it ends up selecting that entire line, but then it places the cursor on the next

Just selecting the entire line also includes a line break character at its end. And the cursor is placed at the end of the selection, i.e. after this character.

**Comment by advice2020** [comment]
@Skif-off 
Thank you for your repsonse
Triple mouse click? 
Yes same result for single, double and triple click.
I might not understand what you mentioned

EXAMPLE:
"I" represents the "cursor"
If I click on the 15 I would like it to do the following
```
14
15 text text text text textI         (this line is selected with cursor at the end)
16
```
but instead it does
```
14
15 text text text text text     (this line is selected)
16 I      (this line gets cursor at the beginning and also gets line highlight color if theme is setup for one)
```

Thank You

**Comment by Skif-off** [comment]
> Triple mouse click? Yes same result for single, double and triple click.

Single mouse click _over the text_ to set the text cursor to the position under the mouse cursor.
Double-click will select the word under the cursor.
Triple-click will select the line under the cursor.
(And this works in a wide variety of programs.)

**Comment by advice2020** [comment]
@Skif-off
What! I did not even know about that.
You are right it works in all the applications and even fields that I have tried this in so far.
I still would like the request I made to be added to Geany, but this is good to know.
Thank you for sharing this :)
